rom/dev/src/error.rs

```diff
@@ -43,7 +43,7 @@ macro_rules! rom_err_def {
         impl From<$enum_name> for core::num::NonZeroU32 {
             fn from(val: $enum_name) -> Self {
                 // Panic is impossible as long as the enums don't define zero.
-                core::num::NonZeroU32::new(((($crate::error::RomComponent::$comp_name) as u32) << 24) | (val as u32)).unwrap()
+                core::num::NonZeroU32::new(((($crate::error::RomComponent::$comp_name) as u32) << 16) | (val as u32)).unwrap()
```

Comment:
Shouldn't this be consistent with this macro in other files such as 

https://github.com/chipsalliance/caliptra-sw/blob/main/drivers/src/error.rs#L76

?

Reply:
I looked at that code. The number of components in the drivers are very small and I thought that 8 bits should be enough for that. Actually, we first started with this way of defining errors from the driver module and carried over all other places and then introduced this error. I believe that drivers should be ok. But please let me know if we want to make it consistent across the board so that it is easier to decipher the error code.

Reply:
I think it needs to be consistent to avoid collisions? 

With your PR as is, InitDevId and DeobfuscationEngine components both have the same 16-bit prefix 0x0100

Reply:
Let me check and change it to 16 bits of component id.

Reply:
The error encoding is going to change in the upcoming error crate.

Reply:
@rusty1968 I think that is the right way to go about this. Else it will be very hard to avoid collisions.